metric_metadata/cassandra/cassandra.go

```diff
@@ -130,42 +129,33 @@ func (db *cassandraDatabase) AddMetricNames(metrics []api.TaggedMetric) error {
 	queryInsert := "INSERT INTO metric_names (metric_key, tag_set) VALUES (?, ?)"
 	queryUpdate := "UPDATE metric_name_set SET metric_names = metric_names + ? WHERE shard = ?"
 
-	c := make(chan *gocql.Query, 10)
-	done := make(chan bool)
-	go func() {
```

Comment:
Entirely possible. This is mostly a failing of the go driver for cassandra. The Java/C# driver handles this correctly for bulk-inserts.